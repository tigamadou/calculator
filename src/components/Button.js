import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';

const Button = ({
  name, handleClick, color, wide,
}) => {
  const [buttonClass, setButtonClass] = useState('button');
  useEffect(() => {
    let btnclass = 'button';
    if (color) {
      btnclass = `${btnclass} white`;
    }

    if (wide) {
      btnclass = `${btnclass} wide`;
    }
    setButtonClass(btnclass);
  }, []);
  return (
    <button type="button" className={buttonClass} onClick={() => handleClick(name)}>
      {name}
    </button>
  );
};

Button.propTypes = {
  name: PropTypes.string,
  handleClick: PropTypes.func,
  color: PropTypes.bool,
  wide: PropTypes.bool,
};

Button.defaultProps = {
  name: null,
  handleClick: null,
  color: null,
  wide: null,
};

export default Button;
