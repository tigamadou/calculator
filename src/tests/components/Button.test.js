import React from 'react';
import { render, screen } from '@testing-library/react';
import Button from '../../components/Button';

describe('Button', () => {
  test('renders Button component ', () => {
    render(<Button />);
    // eslint-disable-next-line no-unused-expressions
    expect(screen.getAllByRole('button')).toBeInTheDocument;
  });
});
