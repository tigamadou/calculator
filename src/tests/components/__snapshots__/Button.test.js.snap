// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component tests It match with snapshot 1`] = `
<button
  className="button"
  onClick={[Function]}
  type="button"
/>
`;
